exp/api/v1beta1/azuremachinepool_default_test.go

```diff
@@ -84,6 +84,18 @@ func TestAzureMachinePool_SetIdentityDefaults(t *testing.T) {
 		SystemAssignedIdentityRole: &infrav1.SystemAssignedIdentityRole{},
 	}}}
 
+	bothFieldsPopulatedTest := test{machinePool: &AzureMachinePool{Spec: AzureMachinePoolSpec{
```

Comment:
I think it would be better to rename this test to `bothRoleAssignmentNamesPopulatedTest` as it is a bit more descriptive!